mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -461,33 +461,39 @@ def _get_job_run_id(self, run):
             return None
         return run.data.tags.get(MLFLOW_DATABRICKS_JOB_RUN_ID, None)
 
+    def _truncate_input_sources(self, run):
+        num_input_sources = len(run.inputs.dataset_inputs)
+        if num_input_sources > _MAX_LINEAGE_DATA_SOURCES:
+            _logger.warning(
+                f"Truncating the number of lineage input sources from "
+                f"{str(num_input_sources)} to 10."
```

Comment:
Could we make this message something that tells the user what's happening in context as they're creating a model version, e.g. `f"Model version has str(num_input_sources)} upstream datasets, which exceeds the max of 10 upstream datasets for lineage tracking. Only the first 10 datasets will be propagated to Unity Catalog lineage"